app/src/ui/history/compare.tsx

```diff
@@ -258,7 +258,14 @@ export class CompareSidebar extends React.Component<
         localCommitSHAs={this.props.localCommitSHAs}
         emoji={this.props.emoji}
         onViewCommitOnGitHub={this.props.onViewCommitOnGitHub}
-        onRevertCommit={this.props.onRevertCommit}
+        onRevertCommit={
+          // do not allow revert if in comparison:behind mode
+          (formState.kind === HistoryTabMode.History
```

Comment:
Can do.  However, any suggestions for a function name or naming, or location?  If it is outside the class, it will require passing arguments in, while inside the class it needs no arguments.

Reply:
i tend to prefer pushing as much logic into pure functions out of the component as possible, but i think having it as a private method of this class would be okay too.

i think the way i'd do it is to have a method outside the class called `ableToRevertCommit` that returns a `boolean`, and then use that `boolean` in the component to control whether the component gets `undefined` or `this.props.onRevertCommit`. or something similar :). that way the business logic is extracted and we keep the component's concerns localized to the component.

Reply:
Thanks for the suggested function name …  I was stuck at just 'a'.  :)